spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -85,67 +242,167 @@ def write_recording(recording, save_path, acquisition_name='ElectricalSeries', *
+                        location='unknown',
+                        filtering='none',
+                        group=grp,
+                    )
+            electrode_table = nwbfile.electrodes
+
+            # add/update electrode properties
+            for ch in channel_ids:
+                rx_channel_properties = recording.get_channel_property_names(channel_id=ch)
+                for pr in rx_channel_properties:
+                    val = recording.get_channel_property(ch, pr)
+                    # group property of electrodes can not be updated
+                    if pr == 'group':
+                        continue
+                    # property 'location' of RX channels corresponds to x, y, z of NWB electrodes
+                    if pr == 'location':
+                        continue
+                    # property 'brain_area' of RX channels corresponds to 'location' of NWB electrodes
+                    if pr == 'brain_area':
+                        pr = 'location'
+                    set_dynamic_table_property(
+                        dynamic_table=electrode_table,
+                        row_ids=[ch],
+                        property_name=pr,
+                        values=[val],
+                        default_value=np.nan,
+                        description='no description'
+                    )
+
+            # Tests if ElectricalSeries already exists in acquisition
+            aux = [isinstance(i, ElectricalSeries) for i in nwbfile.acquisition.values()]
+            if not any(aux):
+                electrode_table_region = nwbfile.create_electrode_table_region(
+                    list(range(n_channels)),
+                    'electrode_table_region'
+                )
+                rate = recording.get_sampling_frequency()
+                if 'gain' in recording.get_shared_channel_property_names():
+                    gains = np.array(recording.get_channel_gains())
+                else:
+                    gains = np.ones(n_channels)
+
+                def data_generator(recording, num_channels):
+                    #  generates data chunks for iterator
+                    for id in range(0, num_channels):
+                        data = recording.get_traces(channel_ids=[id]).flatten()
+                        yield data
+
+                data = data_generator(recording=recording, num_channels=n_channels)
+                ephys_data = DataChunkIterator(data=data, iter_axis=1)
+                acquisition_name = 'ElectricalSeries'
+
+                # If traces are stored as 'int16', then to get Volts = data*channel_conversion*conversion
+                ephys_ts = ElectricalSeries(
+                    name=acquisition_name,
+                    data=ephys_data,
+                    electrodes=electrode_table_region,
+                    starting_time=recording.frame_to_time(0),
+                    rate=rate,
+                    conversion=1e-6,
+                    channel_conversion=gains,
```

Comment:
we should only include this if they are not all ones

Reply:
With that I just noticed the `gain` property is being stored both in the electrodes_table and in the ElectricalSeries. I think it would be better to store it in the electrodes_table, where it could be updated if needed.
What do you think @bendichter ?

Reply:
for now: 2d17a2fd629c2656716a24543a779dd24aed40c0

but I think we could remove this part altogether and leave it as a column of electrodes_table

Reply:
The potential in volts = `ElectricalSeries.data` * `ElectricalSeries.conversion` [scalar applied to all channels] * `ElectricalSeries.channel_conversion` [vector applied to individual channels]. This information should *not* be in the electrodes table in an NWB file.

Reply:
yes, that makes sense. I removed the gains from the electrodes_table, it is now saved in the conversion fields of the ElectricalSeries, if the ElectricalSeries is being created. Important to note, this cannot be changed later by the user since these fields cannot be changed.
535ee0613491f73ee2896d1080d5067b47d8dc23